packages/story-editor/src/components/form/tags/input.js

```diff
@@ -110,8 +181,12 @@ function Input({
           if (e.key === 'Backspace' && e.target.value === '') {
             dispatch({ type: ACTIONS.REMOVE_TAG });
           }
-          if (['Comma', 'Enter', 'Tab'].includes(e.key)) {
-            dispatch({ type: ACTIONS.SUBMIT_VALUE });
+          if (['Comma', ',', 'Enter', 'Tab'].includes(e.key)) {
+            e.preventDefault();
+            e.stopPropagation();
+
+            await dispatch({ type: ACTIONS.SUBMIT_VALUE });
+            await inputRef?.current.focus();
```

Comment:
I'll retest this, because clearly i'm off this week, but this was to make sure the value was submitted and THEN the input refocused to restart the reducer state with 'onFocus' without a value present as input.

Reply:
ah yea. tough race case. I don't think the `await` s there actually change any behavior, but I think the `dispatch({ type: ACTIONS.SUBMIT_VALUE })` _should_ always get called and update state before any dispatches called by `onFocus`

My rationale for that is that it will have fired off the `dispatch({ type: ACTIONS.SUBMIT_VALUE });` before the `onFocus` event gets put in the event queue, and react wont batch the `dispatch` calls unless they're in the same event handler callback. That's at least my best guess. so it shouldn't change anything removing those `await` s and it should still function how you're envisioning. but that being said, I'm not 100% sure